# 16s analysis 

[TOC]

# What is Qiime2?

QIIME (Quantitative Insights Into Microbial Ecology) is a powerful, extensible, and decentralized microbiome analysis package with a focus on data and analysis transparency. QIIME 2 is a major revision of the original QIIME package.  QIIME 2 enables researchers to start an analysis with raw DNA sequence data and finish with publication-quality figures and statistical results.  The acronym is canonically pronounced, "chime" though some people say "kime" (and that's ok).  We usually use ALL CAPS and a space between QIIME and 2 when discussing the package, and "qiime" or "qiime2" when talking about invoking specific programs.

Key features include:

* Integrated and automatic tracking of data provenance 
* Semantic type system
* Plugin system for extending microbiome analysis functionality 
* Support for multiple types of user interfaces (e.g. API, command line, graphical)

We are using qiime2-2024.2.  The latest release is qiime2-2024.2.  You can read about the core concepts of QIIME 2 [at the QIIME website.](https://docs.qiime2.org/2024.2/concepts/)

## Key vocabulary. 

![](https://i.imgur.com/x8iWgyI.png)


### Data files: QIIME 2 artifacts
Data produced by QIIME 2 exist as QIIME 2 artifacts. 

A QIIME 2 artifact contains **data and metadata**. The metadata describes things about the data, such as its type, format, and how it was generated (provenance). A QIIME 2 artifact typically has the `.qza` file extension when stored in a file.

Since QIIME 2 works with artifacts instead of data files the first thing to do is to import your data. Typically you will start by importing raw sequence data. 


### Data files: visualizations
Visualizations are another type of data generated by QIIME 2. When written to disk, visualization files typically have the `.qzv` file extension. 


Use https://view.qiime2.org to easily view QIIME 2 visualizations files. 
Open the website, transfer the file to your computer, and drag and drop!


## Prepare files for Qiime2


### Fastq files: 
Your data as fastq or fastq.gz files
### Qiime2 files
Now, we need to prepare two files for running the program one lists the samples and tells the program "where" they are (file manifest). 
In this file, the first column is the sample ID, the next is the path to the files. 

The second file (sample metadata) provides the information relevant to the analysis. Here you will list things like the treatments, time in culture, type of body part, or any variable that can be relevant to interpreting the data. In the first column we will list the sample ids (this has to match the ID in the manifest file). Then, each column is a different variable. 
The first line names the variables, and the second line of the file lets the program if we are listing into a categorical or numerical variable. For a categorical column, *you should identify the column as categorical*, and for numerical, *identify the column as numeric, and include only numbers and empty cells!*. 

[metadata](https://earthmicrobiome.org/protocols-and-standards/metadata-guide/)  

#### Where are your samples?: “Fastq manifest” formats

Import your data into QIIME 2 manually by first creating a “manifest file” 

1.  You’ll create a text file called a manifest file named `pe-33-manifest.txt`, which maps sample identifiers to fastq files (or fastq.gz). 

2. The manifest file also indicates the direction of the reads (R1, R2).

3. The manifest file is a tab-separated (i.e., .tsv) text file. The first column defines the Sample ID, while the second (and optional third) column defines the absolute file path to the forward (and optional reverse) reads. All of the rules and behavior of this format are inherited from the QIIME 2 Metadata format.


So my file will look like

:::info
sample-id	forward-absolute-filepath	reverse-absolute-filepath
sampleA	/path/filenameA_1.fastq.gz    /path/filenameA_2.fastq.gz
sampleB	/path/filenameB_1.fastq.gz	/path/filenameB_2.fastq.gz
:::

Once you have all the info in a txt file on your computer, we are ready to create the file on the server. 


1. You can transfer the file using FileZilla. 
2. you can create the file using the command line. 
To create the file, type in the terminal to create the txt. file. 

```
cat > pe-33-manifest.txt
``` 
And now let's populate it with the info you have prepared. 

* Copy from the file on your computer and paste it into the terminal. Press enter to add an extra line.
* Close the file using Ctrl + Z

Let's check all looks correct!
```
ls -l pe-33-manifest.txt
head pe-33-manifest.txt
tail pe-33-manifest.txt
```

note. head and tail will by default give you 10 lines. If you want to see more use the flag number of lines `-n ` for example `-n 100` will list 100 lines. 

If something went wrong you can remove the file and start over!
```
rm pe-33-manifest.txt

```

### Sample metadata 
This file summarizes the characteristics of your samples. Open the file in excel and select the columns you are interested in. Then just format it as Qiime2 and save it in a txt file on your computer (name it sample-metadata.tsv). 

As we did with the manifest file, once you have all the info in a txt file on your computer, we are ready to create the file on the server. 


1. You can transfer the file using FileZilla. 
2. you can create the file using the command line. 

As we did before, create a file 
```
cat > sample-metadata.tsv
```
And now let's populate it with the info you have prepared. 

* Enter your document's text. Copy from the file on your computer and paste it into the terminal. Press enter to add an extra line.
* Close the document using Ctrl + Z
and check!
```
ls -l sample-metadata.tsv
head sample-metadata.tsv
tail sample-metadata.tsv
```

:::info
SampleID	name	type	date	trip	site	fishID	site.trip
#q2:types	categorical	categorical	categorical	categorical	categorical	categorical	categorical
a31	a31	Gut	5/3/22	2	Chitili	160	Chitili.Spring
a32	a32	Gut	5/3/22	2	Chitili	169	Chitili.Spring
a33	a33	Gut	5/3/22	2	Chitili	171	Chitili.Spring
:::
if something went wrong you can remove the file and start over!
```
rm sample-metadata.tsv
```


# Running Qiime2

Let's activate the program!

Because Qiime2 is a suite of programs that call a lot of other programs it requires a lot of modifications to the system to run properly.  So we will be running Qiime2 in a special contained environment called conda.  Initializing the conda environment and starting Qiime2 is a multi-step process that varies depending on the specific computer architecture, so we've tried to make this all happen in the background by creating two simple commands to start and stop qiime.
To start qiime2:  

```
   conda activate qiime2-amplicon-2024.10   
```
to exit

```
conda deactivate
```

This deactivates the conda-qiime environment and you will see that you no longer have '(qiime)' preceding your prompt.

# Importing data into Qiime2 and generating feature tables

This flowchart describes all demultiplexing steps that are currently possible in QIIME 2, depending on the type of raw data you have imported. Usually only one of the different demultiplexing actions available in q2-demux or q2-cutadapt will apply to your data, and that is all you will need. In this tutorial, we will use demultiplexed paired-end reads stored in fastq files. We will merge the reads and denoise the data using Dada2. 

![](https://i.imgur.com/SwxRkwA.png)


## Data import

![](https://i.imgur.com/cjleTz9.png)

Using the manifest file we import the files of interest into QIIME2. These sequences are already demultiplexed, so it will import as "demux.qza"


:::info
**how you read the command**

*what* .... qiime tools import  
*on what type of file are my sequences?*   .... --type 'SampleData[PairedEndSequencesWithQuality]'   
*where are the sequences*  ...  --input-path pe-33-manifest.txt  
*How they look* --input-format PairedEndFastqManifestPhred33V2  
*Where do I want my outputs* ... --output-path paired-end-demux.qza  
:::

:::success
qiime tools import \
&nbsp;    --type 'SampleData[PairedEndSequencesWithQuality]' \
&nbsp;    --input-path pe-33-manifest.txt \
&nbsp;    --output-path paired-end-demux.qza \
&nbsp;    --input-format PairedEndFastqManifestPhred33V2 
:::

Run in terminal

```
qiime tools import --type 'SampleData[PairedEndSequencesWithQuality]' --input-path pe-33-manifest.txt --output-path paired-end-demux.qza  --input-format PairedEndFastqManifestPhred33V2
```

Now we are going to generate a human-readable summary (we will do this after many steps, first we compute the data, then we summarize)
:::success
qiime demux summarize \
&nbsp;    --i-data paired-end-demux.qza  \
&nbsp;    --o-visualization paired-end-demux.qzv
:::

Run in the terminal
```
qiime demux summarize --i-data paired-end-demux.qza --o-visualization paired-end-demux.qzv
```

Using human-readable or the transfer command in your terminal (scp), drop the file qzv into your local computer. 
You can visualize the results in https://view.qiime2.org/ 

--------
  


![](https://i.imgur.com/Z6Be0ec.png)

See: https://docs.qiime2.org/



https://forum.qiime2.org/t/tutorial-integrating-qiime2-and-r-for-data-visualization-and-analysis-using-qiime2r/4121 


